src/physics/utilities/equation_solver.cpp

```diff
@@ -58,11 +58,38 @@ std::unique_ptr<mfem::IterativeSolver> EquationSolver::buildIterativeLinearSolve
 }
 
 std::unique_ptr<mfem::NewtonSolver> EquationSolver::buildNewtonSolver(MPI_Comm                         comm,
-                                                                      const NonlinearSolverParameters& nonlin_params,
-                                                                      mfem::Solver&                    lin_solver)
+                                                                      const NonlinearSolverParameters& nonlin_params)
 {
-  auto newton_solver = std::make_unique<mfem::NewtonSolver>(comm);
-  newton_solver->SetSolver(lin_solver);
+  std::unique_ptr<mfem::NewtonSolver> newton_solver;
+
+  if (nonlin_params.nonlin_solver == NonlinearSolver::BasicNewton) {
+    newton_solver = std::make_unique<mfem::NewtonSolver>(comm);
+  }
+  // KINSOL
+  else {
+#ifdef MFEM_USE_SUNDIALS
+    auto kinsol_strat = KIN_NONE;
+    switch (nonlin_params.nonlin_solver) {
+      case NonlinearSolver::KINBasic:
```

Comment:
`KIN_NONE` is "A full newton step", `KIN_LINESEARCH` is backtracking line search. Food for more expressive naming.